C++/test/data_structure/stack/stack.cpp

```diff
@@ -0,0 +1,26 @@
+#include "data_structure/stack/stack.hpp"
+#include "third_party/catch.hpp"
+
+TEST_CASE("Verify initial state", "[stack]") {
```

Comment:
This is a pretty small testing file. Look at `singly_linked_list.cpp` for a guide of how to expand it.